File: backend/ScolabBackend/src/main/java/com/Scolab/ScolabBackend/Entity/Tache.java
package com.Scolab.ScolabBackend.Entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.time.LocalDateTime;
import java.util.List;

@Document(collection = "taches")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Tache {
    @Id
    private String id;

    @Field("titre")
    private String titre;

    @Field("description")
    private String description;

    @Field("priorite")
    private Priorite priorite = Priorite.MOYENNE;

    @Field("statut")
    private StatutTache statut = StatutTache.A_FAIRE;

    @Field("date_debut")
    private LocalDateTime dateDebut;

    @Field("date_echeance")
    private LocalDateTime dateEcheance;

    @Field("date_fin")
    private LocalDateTime dateFin;

    @DBRef
    @Field("assignes_a")
    private List<Etudiant> assignesA;

    @Field("projet_id")
    private String projetId;

    @DBRef
    @Field("livrable_associe")
    private Livrable livrableAssocie;

    // Méthodes métier
    public boolean estEnRetard() {
        return LocalDateTime.now().isAfter(dateEcheance) && statut != StatutTache.TERMINEE;
    }

    public boolean isTerminee() {
        return statut == StatutTache.TERMINEE;
    }

    public void changerStatut(StatutTache nouveauStatut) {
        this.statut = nouveauStatut;
        if (nouveauStatut == StatutTache.TERMINEE) {
            this.dateFin = LocalDateTime.now();
        }
    }
}

File: app/src/main/java/com/example/scolabstudentapp/models/Tache.kt
package com.example.scolabstudentapp.models

import com.google.gson.annotations.SerializedName

data class Tache(
    @SerializedName("_id")
    val id: String,
    @SerializedName("titre")
    val titre: String,
    @SerializedName("description")
    val description: String?,
    @SerializedName("dateEcheance")
    val dateEcheance: String,
    @SerializedName("statut")
    val statut: String, // Ex: "A_FAIRE", "EN_COURS"
    @SerializedName("priorite")
    val priorite: String,
    @SerializedName("projetId")
    val projetId: String,
    @SerializedName("assignesA")
    val assignesA: List<User>? = null
)
